src/lib/truth/truth.ml

```diff
@@ -2,8 +2,10 @@ type true_ = unit
 
```

Comment:
Is there a reason we can't make new types for true_ and false_? Like

```
type true_ = True
type false_ = False
```

It's a little scary them both being the same, I know the mli keeps them abstract but we might as well have this as well I think.

Reply:
I think we just create two types with no inhabitance since they are used as phantom types.

Reply:
Yes, I'm not sure why I did it this way originally. I think I just wasn't familiar with the `type true_` syntax for declaring types without inhabitants. I made sure to hide the types in the `mli` though.